### PR TITLE
Add ccp

### PR DESCRIPTION
**Pull request by alexgraham-da**
Add CentralCounterParty and CCPCustomer. CCPCustomer maintains margin and clearing accounts with the CCP who can perform margin calls, mark to market calculations, and novate trades.

### CODE REVIEW

File: daml/Marketplace/Derivative.daml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+template Derivative
+  with
+    id : Id
+    pricePrecision : Int
+    description : Text
+    prodSym : Text  -- BTCF21
+    prodCode : Text  -- BTC
+    uuid : Text
+    secType : SecType
+    mmy : Text  -- YYYYMM or YYYYMMDD or YYYMMwN
```
Comment (alexmatson-da) [inline daml/Marketplace/Derivative.daml:37]: why not a `Date`?

**Review by alexmatson-da — APPROVED** [review]
Looks good!

File: daml/Marketplace/Derivative.daml (hunk 1 above)
Comment (alexgraham-da) [reply daml/Marketplace/Derivative.daml:37]: Dimitri added this, but I'm pretty sure it's the specified date format as opposed to the actual date

Comment (alexmatson-da) [reply daml/Marketplace/Derivative.daml:37]: Oh, I see. We might want to make it an enum or something if that's the case, but this field also doesn't seem to be in use anywhere, so we can leave it be as well

Comment (alexgraham-da) [reply daml/Marketplace/Derivative.daml:37]: Yeah agreed. I'll do some digging to see if it will be used and how, but we'll leave it for now.

**Pull request merged**